### PR TITLE
Add sample file for the official documentation

### PR DESCRIPTION
**Pull request by ppamorim**
This PR will close the issue #11.
@bidoubiwa Could you please check if the file is correct? It is not completed yet but I think you can review it anyway. :)

### CODE REVIEW

**Comment by bidoubiwa** [comment]
The file format is good, the file name is good and the location is also good!

**Comment by curquiza** [comment]
@ppamorim, thanks a lot for your PR 😁

A small question.
Why some functions has to be used like this:

```swift
...
switch result {
      case .success(let index):
          print(index)
      case .failure(let error):
          print(error)
      }
```

and some other like this
```swift
... { return in }
```

**Comment by ppamorim** [comment]
@curquiza I will migrate all cases to use the first example, it's more clear for the user since we are defining the types explicitly.

**Comment by curquiza** [comment]
@ppamorim I agree with that, since the examples are consistent!
Thanks for taking the time to do these docs, it's really useful!!!

**Comment by ppamorim** [comment]
@curquiza Waiting for the review of the #16, that's needed since some documentation is missing.

File: .code-samples.meilisearch.yaml [hunk 1]
```diff
@@ -0,0 +1,866 @@
...
+  client.getSynonyms(UID: "movies") { (result: Result<[String: [String]], Swift.Error>) in
+      switch result {
+      case .success(let synonyms):
+          print(update)
+      case .failure(let error):
+          print(error)
+      }
+  }
+update_synonyms_1: |-
+  let synonym = Synonym()
```
Comment (curquiza) [inline .code-samples.meilisearch.yaml:189]: Can you add an example of synonyms? :) 
Like:
```json
{
  "wolverine": ["xmen", "logan"],
  "logan": ["wolverine", "xmen"],
  "wow": ["world of warcraft"]
}
```

File: .code-samples.meilisearch.yaml [hunk 2]
```diff
@@ -0,0 +1,866 @@
...
+  client.getRankingRules(UID: "movies") { (result: Result<[String], Swift.Error>) in
+      switch result {
+      case .success(let rankingRules):
+          print(rankingRules)
+      case .failure(let error):
+          print(error)
+      }
+  }
+update_ranking_rules_1: |-
+  let rankingRules: [String] = ["of", "the", "to"]
```
Comment (curquiza) [inline .code-samples.meilisearch.yaml:245]: There are not ranking rules 😉

File: .code-samples.meilisearch.yaml [hunk 3]
```diff
@@ -0,0 +1,866 @@
...
+  client.getSetting(UID: "movies") { (result: Result<Setting, Swift.Error>) in
+      switch result {
+      case .success(let setting):
+          print(setting)
+      case .failure(let error):
+          print(error)
+      }
+  }
+update_settings_1: |-
+  let settings = Settings(...)
```
Comment (curquiza) [inline .code-samples.meilisearch.yaml:161]: Can you add the same example https://docs.meilisearch.com/references/settings.html#example-2 ? 🙂

**Comment by curquiza** [comment]
Hey @ppamorim, can you add this comment at the top of the code-sample file? 🙂

```yml
# This code-samples file is used by the MeiliSearch documentation
# Every example written here will be automatically fetched by
# the documentation on build
# You can read more on https://github.com/meilisearch/documentation/tree/master/.vuepress/code-samples
---
```

**Comment by ppamorim** [comment]
> Hey @ppamorim, can you add this comment at the top of the code-sample file? 🙂
> 
> ```yaml
> # This code-samples file is used by the MeiliSearch documentation
> # Every example written here will be automatically fetched by
> # the documentation on build
> # You can read more on https://github.com/meilisearch/documentation/tree/master/.vuepress/code-samples
> ---
> ```

I will resume the work of this PR this week(end).

File: .code-samples.meilisearch.yaml (hunk 2 above)
Comment (ppamorim) [reply .code-samples.meilisearch.yaml:245]: I think I copied them from the javascript library, now I am confused. What I should do?

Comment (curquiza) [reply .code-samples.meilisearch.yaml:284]: Oh! This probably was a mistake when you copied. Now it's fix, see [these lines](https://github.com/meilisearch/meilisearch-js/blob/de9f211fbf1987e2218f25740894d52fc2becaa4/.code-samples.meilisearch.yaml#L110-L120).

**Comment by bidoubiwa** [comment]
Hey @ppamorim, Is this still a draft? Should I review it as if it was a PR?

**Comment by ppamorim** [comment]
@bidoubiwa Yes, I am waiting the facets document gets merged.

File: .code-samples.meilisearch.yaml [hunk 4]
```diff
@@ -0,0 +1,914 @@
...
+      case .success(let update):
+          print(update)
+      case .failure(let error):
+          print(error)
+      }
+  }
+add_movies_json_1: |-
+  let path = Bundle.main.url(forResource: "movies", withExtension: "json")
+  let document: Data = Data(contentsOf: path)
+  client.addOrUpdateDocument(UID: "movies", document: document) { (result: Result<Update, Swift.Error>) in
```
Comment (curquiza) [inline .code-samples.meilisearch.yaml:763]:
```suggestion
  let documents: Data = Data(contentsOf: path)
  client.addDocuments(UID: "movies", documents: documents) { (result: Result<Update, Swift.Error>) in
```

File: .code-samples.meilisearch.yaml [hunk 5]
```diff
@@ -0,0 +1,914 @@
...
+  }
+documents_guide_add_movie_1: |-
+  let documentJsonString = """
+  [{
+      "movie_id": 123sq178,
+      "title": "Amélie Poulain"
+  }]
+  """
+  let document: Data = documentJsonString.data(using: .utf8)!
+  client.addOrUpdateDocument(UID: "movies", document: document) { (result: Result<Update, Swift.Error>) in
```
Comment (curquiza) [inline .code-samples.meilisearch.yaml:779]:
```suggestion
  let documents: Data = documentJsonString.data(using: .utf8)!
  client.addDocuments(UID: "movies", documents: documents) { (result: Result<Update, Swift.Error>) in
```

File: .code-samples.meilisearch.yaml [hunk 6]
```diff
@@ -0,0 +1,914 @@
...
+  ```
+    dependencies: [
+      .package(url: "https://github.com/meilisearch/meilisearch-swift.git")
+    ]
+  ```
+  Example:
+  ```swift
+    let client = try! MeiliSearch(Config(hostURL: "http://localhost:7700"))
+    let document: Data = /...
+    client.addOrUpdateDocument(UID: "movies", document: document) { (result: Result<Update, Swift.Error>) in
```
Comment (curquiza) [inline .code-samples.meilisearch.yaml:842]:
```suggestion
    let documents: Data = /...
    client.addDocuments(UID: "movies", documents: documents) { (result: Result<Update, Swift.Error>) in
```

File: .code-samples.meilisearch.yaml [hunk 7]
```diff
@@ -0,0 +1,967 @@
...
+          print(index)
+      case .failure(let error):
+          print(error)
+      }
+  }
+update_an_index_1: |-
+  client.updateIndex(UID: "movies") { (result: Result<(), Swift.Error>) in
+      switch result {
+      case .success:
+          print("Index updated")
```
Comment (bidoubiwa) [inline .code-samples.meilisearch.yaml:37]:
```suggestion
  client.updateIndex(UID: "movies") { (result: Result<(Index), Swift.Error>) in
      switch result {
      case .success (let index):
          print(index)
```

As per the [documentation](https://docs.meilisearch.com/references/indexes.html#update-an-index)

File: .code-samples.meilisearch.yaml [hunk 8]
```diff
@@ -0,0 +1,967 @@
...
+get_keys_1: |-
+    let masterKey = "c4dff5a196824a0704c2023916c21eaf0a13485a9d637416820bc623375f2a"
+    client.keys(masterKey: masterKey) { (result: Result<Key, Swift.Error>) in
+        switch result {
+        case .success(let key):
+            print(key)
+        case .failure(let error):
+            print(error)
+        }
+    }
```
Comment (bidoubiwa) [inline .code-samples.meilisearch.yaml:155]: Isn't this function suppose to be accessible based on the `client` key without needing a key to be given in parameter?

File: .code-samples.meilisearch.yaml [hunk 9]
```diff
@@ -0,0 +1,967 @@
...
+      case .success(let searchResult):
+          print(searchResult)
+      case .failure(let error):
+          print(error)
+      }
+  }
+faceted_search_walkthrough_facets_distribution_1: |-
+  let searchParameters = SearchParameters(
+      query: "Batman",
+      facetsDistribution: ["genres"]])
```
Comment (bidoubiwa) [inline .code-samples.meilisearch.yaml:959]:
```suggestion
  let searchParameters = SearchParameters(
      query: "Batman",
      facetsDistribution: ["genres"])
```

File: .code-samples.meilisearch.yaml (hunk 7 above)
Comment (ppamorim) [reply .code-samples.meilisearch.yaml:37]: I need to update the client.

File: .code-samples.meilisearch.yaml (hunk 8 above)
Comment (ppamorim) [reply .code-samples.meilisearch.yaml:155]: This is probably outdated.

**Pull request merged**